File: backend/src/main/java/com/pharmacy/pharmacy/stock/StockInfo.java
package com.pharmacy.pharmacy.stock;

import com.pharmacy.pharmacy.batch.MedicineBatch;

import java.util.List;

/**
 * DTO that presents stock information for a medicine.
 * Combines medicine details with aggregated batch/stock data.
 */
public class StockInfo {

    private Long medicineId;
    private String medicineName;
    private String category;
    private int totalStock;
    private List<MedicineBatch> batches;

    public StockInfo() {
    }

    public StockInfo(Long medicineId, String medicineName, String category, int totalStock, List<MedicineBatch> batches) {
        this.medicineId = medicineId;
        this.medicineName = medicineName;
        this.category = category;
        this.totalStock = totalStock;
        this.batches = batches;
    }

    // Getters and setters

    public Long getMedicineId() {
        return medicineId;
    }

    public void setMedicineId(Long medicineId) {
        this.medicineId = medicineId;
    }

    public String getMedicineName() {
        return medicineName;
    }

    public void setMedicineName(String medicineName) {
        this.medicineName = medicineName;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public int getTotalStock() {
        return totalStock;
    }

    public void setTotalStock(int totalStock) {
        this.totalStock = totalStock;
    }

    public List<MedicineBatch> getBatches() {
        return batches;
    }

    public void setBatches(List<MedicineBatch> batches) {
        this.batches = batches;
    }
}

File: frontend/lib/features/stock/models/stock_info.dart
import '../../batches/models/medicine_batch.dart';

class StockInfo {
  final int medicineId;
  final String medicineName;
  final String category;
  final int totalStock;
  final List<MedicineBatch> batches;

  StockInfo({
    required this.medicineId,
    required this.medicineName,
    required this.category,
    required this.totalStock,
    required this.batches,
  });

  factory StockInfo.fromJson(Map<String, dynamic> json) {
    return StockInfo(
      medicineId: json['medicineId'] as int,
      medicineName: json['medicineName'] as String,
      category: json['category'] as String,
      totalStock: (json['totalStock'] as num).toInt(),
      batches: (json['batches'] as List?)
              ?.map((e) => MedicineBatch.fromJson(e))
              .toList() ??
          [],
    );
  }
}
